application/src/main/java/org/togetherjava/tjbot/commands/tags/TagSystem.java

```diff
@@ -140,14 +139,16 @@ void putTag(String id, String content) {
      *
      * @return a set of all ids known to the system, not backed
      */
-    Set<String> getAllIds() {
+    List<String> getAllIds() {
         return database.readTransaction(context -> {
             try (var select = context.select(Tags.TAGS.ID)) {
                 return select.from(Tags.TAGS)
                     .fetch()
                     .stream()
                     .map(dbRecord -> dbRecord.getValue(Tags.TAGS.ID))
-                    .collect(Collectors.toSet());
+                    .distinct()
+                    .sorted()
+                    .toList();
             }
```

Comment:
From a design-pov, the sorting should not be done here but at the point where it is actually used. The system should still return a `Set<String>` and not a sorted distinct list.